File: sample/app/src/main/java/com/github/kilnn/wristband2/sample/syncdata/db/converter/UUIDConverter.java
package com.github.kilnn.wristband2.sample.syncdata.db.converter;

import android.arch.persistence.room.TypeConverter;

import java.util.UUID;

public class UUIDConverter {

    @TypeConverter
    public static String fromUUID(UUID uuid) {
        return uuid.toString();
    }

    @TypeConverter
    public static UUID fromStr(String str) {
        return UUID.fromString(str);
    }
}

File: sample/app/src/main/java/com/topstep/fitcloud/sample2/utils/room/UUIDConverter.kt
package com.topstep.fitcloud.sample2.utils.room

import androidx.room.TypeConverter
import java.util.*

object UUIDConverter {
    @TypeConverter
    fun fromUUID(uuid: UUID): String {
        return uuid.toString()
    }

    @TypeConverter
    fun fromStr(str: String?): UUID {
        return UUID.fromString(str)
    }
}